File: dotNet/astenn/astenn-core/src/Org/Lestr/Astenn/Configuration/IPermissionsManager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Org.Lestr.Astenn.Configuration
{


    public interface IPermissionsManager {


        List<IPermissionsManagerListener> Listeners { get; }

        
        bool AutoExposeLocalPlugins { get; set; }


        void RescanAutoExposedLocalPlugins();


        void ExposeLocalPlugin(Type localPluginInterface,
                               Type localPluginImplementation);


        void UnexposeLocalPlugin(Type localPluginInterface,
                                 Type localPluginImplementation);


        IEnumerable<Type[]> ExposedLocalPlugins { get; }


    }// END Interface IPermissionsManager


    public interface IPermissionsManagerListener
    {


        void LocalPluginExposed(Type localPluginInterface,
                                Type localPluginImplementation);


        void LocalPluginUnexposing(Type localPluginInterface,
                                   Type localPluginImplementation);


    }// END Interface IPermissionsManagerListener


}// END Namespace Org.Lestr.Astenn.Configuration

File: trunk/astenn-core/src/main/java/org/lestr/astenn/configuration/IPermissionsManager.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package org.lestr.astenn.configuration;

import java.util.Collection;

/**
 *
 * @author PIBONNIN
 */
public interface IPermissionsManager {


    Collection<IPermissionsManagerListener> getListeners();


    void setAutoExposeLocalPlugins(boolean autoExpose);


    boolean isAutoExposeLocalPlugins();


    void rescanAutoExposedLocalPlugins();


    void exposeLocalPlugin(Class<?> localPluginInterface,
                           Class<?> localPluginImplementation);


    void unexposeLocalPlugin(Class<?> localPluginInterface,
                             Class<?> localPluginImplementation);


    Iterable<Class<?>[]> getExposedLocalPlugins();


    public static interface IPermissionsManagerListener {


        void localPluginExposed(Class<?> localPluginInterface,
                                Class<?> localPluginImplementation);


        void localPluginUnexposing(Class<?> localPluginInterface,
                                   Class<?> localPluginImplementation);


    }// END Interface IPermissionsManagerListener


}// END Interface IPermissionsManager


